net/FlatBuffers/Table.cs

```diff
@@ -83,15 +84,15 @@ protected int __vector(int offset)
         }
 
         // Initialize any Table-derived type to point to the union at the given offset.
-        protected TTable __union<TTable>(TTable t, int offset) where TTable : Table
+        public T __union<T>(int offset) where T : struct, Accessor
```

Comment:
Not sure about your use of interfaces, haven't looked into it yet - but I did get chance to look at the benchmarking.

Your change does have an impact on performance (code pushed to here - https://github.com/evolutional/flatbuffers/tree/cs-benchmark)

From what I can tell, the 'encode' test is slightly slower and the 'decode' test is much faster. Both of these have a lower std deviation, which seems to indicate that performance is more consistent - at least in the stuff the benchmark is doing.

Before:

    Name                                  Mean   StdD  Unit
    Encode                               3.853 1.190 us/op
    Decode                               0.416 1.701 us/op
    Traverse                             2.428 0.389 us/op


After:

    Name                                  Mean   StdD  Unit
    Encode                               4.006  0.952 us/op
    Decode                               0.090  0.139 us/op
    Traverse                             2.224  0.426 us/op

This benchmark is fairly old now, ported from the Java one - it'd be a good exercise to bring them all in line with your C++ one to get a true comparison.

It does seem that using structs here results in faster decode performance; the consistency boost is very likely from the reduced garbage collection overhead.

Note that it was a breaking change, however - I had to modify any code which accessed a struct to get the `.Value` member.